src/parser.ts

```diff
@@ -117,9 +121,9 @@ export class Parser extends DiagnosticEmitter {
   ): void {
     var normalizedPath = normalizePath(path);
     var internalPath = mangleInternalPath(normalizedPath);
-
     // check if already processed
     if (this.donelog.has(internalPath)) return;
+    this.currentFile = internalPath.endsWith("/index") ? internalPath.substring(0, internalPath.indexOf("/index")): internalPath;
```

Comment:
This looks like it might be problematic in that it can lead to confusion between `file.ts` and `file/index.ts`, which may co-exist. This is currently being prevented in that the backlog contains the exact request, while asc then provides the correct path when parsing. So, at this stage, `path` should be the correct path.